src/core/src/sys/compute.rs

```diff
@@ -179,19 +181,23 @@ impl Compute for Virial {
     type Output = Matrix3;
     fn compute(&self, system: &System) -> Matrix3 {
         assert!(!system.cell().is_infinite(), "Can not compute virial for infinite cell");
-        let mut virial = Matrix3::zero();
-        for i in 0..system.size() {
+
+        let mut virial = (0..system.size()).into_par_iter().map(|i| {
+            let mut local_virial = Matrix3::zero();
+
             for j in (i+1)..system.size() {
                 let distance = system.bond_distance(i, j);
                 for potential in system.pair_potentials(i, j) {
                     let info = potential.restriction().information(distance);
                     if !info.excluded {
                         let d = system.nearest_image(i, j);
-                        virial += info.scaling * potential.virial(&d);
+                        local_virial += info.scaling * potential.virial(&d);
                     }
                 }
             }
-        }
+
+            local_virial
+        }).reduce(|| Matrix3::zero(), |a, b| a + b);
```

Comment:
Because `iter::Sum` was implemented in #153, now that it's merged I'll rebase it.